icefoganalytics/travel-authorization
# Ytgov-112: Travel Port Text Pop Up and Button

Fixes https://github.com/ytgov/travel-authorization/issues/112

Relates to:

- Travel Desk Request
- Travel options

# Context

> **Is your feature request related to a problem? Please describe.** 
> Related to the "Travel Port Text" button and subsequent pop-up 
![image](https://github.com/user-attachments/assets/87d7a1e8-d602-4645-adc8-d7824a0879ae)
> 
> 1.) Update titles to read horizontally 
![image](https://github.com/user-attachments/assets/824f2d52-6cc4-4d4c-abf5-98b542900ec2)
> 
> 2.) Clean and separate options button does not work 
![image](https://github.com/user-attachments/assets/c6fd1f1f-052b-409d-a271-cf3eb78a2319)
> 
> **Describe the solution you'd like** 
> The user would copy some text from a different application and the Clean and separate options would parse the various components of the text into flight segments. That would appear in the flight segment region. The user could then group segments together into groupings and give the grouping a cost. Grouped segments would appear on the travel desk request screen.
![image](https://github.com/user-attachments/assets/5f7f5feb-8117-4994-8212-a0cf963115e4)
![image](https://github.com/user-attachments/assets/caddfc48-42cd-441d-9b5a-6947fb120d5b)
![image](https://github.com/user-attachments/assets/94fb52a8-a468-45e6-aae7-79edd5c5310b)
> 

**Additional context** 
See sample text.
<details><summary>parsable-text.txt</summary>
<p>

```
Passenger:
ignore this info

Flights:
Air Canada AC303
Departure: 13 Jan 09:05 Pierre Elliott Trudeau Intl Arpt (YUL) Terminal:
Arrival: 13 Jan 11:53 Vancouver Intl Arpt (YVR) Terminal: M
Duration: 5 Hour(s) 48 Minutes
Status: Confirmed
Class: Q
Air Canada AC304
Departure: 13 Jan 09:05 Pierre Elliott Trudeau Intl Arpt (YUL) Terminal:
Arrival: 13 Jan 11:53 Vancouver Intl Arpt (YVR) Terminal: M
Duration: 5 Hour(s) 48 Minutes
Status: Confirmed
Class: Q

Hotels:
Hotel details here...

Cars:
Car rental details here...
```

<p>
</details>

# Implementation

1. Move away from unstable title card component, to standard Vuetify card component.
2. Add minimal test for single example of parsing travel port(al?) text.
3. Add front-end test setup.

## Concerns

1. The current implementation of the travel text parser is virtually impossible to read or debug, and is extremely fragile. A complete re-write is likely necessary before production use, with each new feature backed by multiple test cases.
2. UI is composed of multiple layers of modals, and this violates both accessibility and basic usability considerations. It also makes it very difficult to perform future development and debugging.

# Screenshots

Travel Desk -> Edit (modal) -> Travel Port Text (modal)
![image](https://github.com/user-attachments/assets/3c9896fa-8ca7-4d9a-b294-c95f0dbc4232)
![image](https://github.com/user-attachments/assets/78690c80-f853-4f7e-a79d-4459e4e2fa6b)
![image](https://github.com/user-attachments/assets/a404739b-79fa-4c73-98c4-0ec5adb317d6)

# Testing Instructions

1. Run the test suite via `dev test web` (or `dev test_web`)
3. Boot the app via `dev up`
4. Log in to the app at http://localhost:8080
5. Click on "My Travel Requests" from the top nav drop down.
6. Create travel request.
7. Generate an estimate.
8. Submit to yourself as the supervisor; this feature should be removed in production.
9. Select "Manger View" from top nav drop down.
10. Approve your request.
11. Select "Travel Desk" from the to drop down nav.
12. Check the box beside your travel request, then click on print.
13. Check that this renders correctly, then exit the modal.
14. Click the "edit" button for a particular travel desk request row to open the travel desk  request edit modal.
15. Click on the "Travel Port(al?) Text" call to action button.
16. This opens the _2nd layer_ modal for managing group/flight options; see concerns.
17. Enter the provide text; note this is the only text that is likely to work; see concerns.
18. Check that the example parses.
